### ISSUE TITLE
GameHub is only showing a few games

### ISSUE THREAD

**Issue by neuromancer** [tkashkin/GameHub#48]
I'm using the last revision of GameHub, previously deleting its cache (`rm -Rf ~/.cache/com.github.tkashkin.gamehub/`), but it will load just a few games:

![screenshot](https://user-images.githubusercontent.com/110782/44631915-1ed85180-a949-11e8-9c86-096a8a212788.png)

I assume that this issue is caused by some incorrect field in some of my windows-only games, since it wasn't happening in the previous revisions, where GameHub wasn't fetching the information of these games.

Unfortunately, the use of `--debug` and gdb backtraces won't show any interesting detail:

```
$ gamehub --debug
Registering Stacktrace 0.9.4
[INFO 15:56:32.771370] Application.vala:154: GameHub version: 0.8.0
[INFO 15:56:32.771387] Application.vala:156: Kernel version: 4.18.4-arch1-1-ARCH
[INFO 15:56:32.830218] app.vala:32: Distro: NAME="Arch Linux"
[DEBUG 15:56:32.831505] Settings.vala:163: Loading settings from schema 'com.github.tkashkin.gamehub.ui'
[DEBUG 15:56:32.861578] Settings.vala:163: Loading settings from schema 'com.github.tkashkin.gamehub.auth.steam'
...
[DEBUG 15:56:33.366344] Settings.vala:163: Loading settings from schema 'com.github.tkashkin.gamehub.paths'
[DEBUG 15:56:33.366489] Settings.vala:163: Loading settings from schema 'com.github.tkashkin.gamehub.paths'
[FATAL 15:56:33.425846] [Json] json_object_get_object_member: assertion 'node != NULL' failed
NULL
An error occured (SIGTRAP)
   The reason is likely an uncaught error.
   Note: no file path and line numbers can be retrieved. Are you sure you added -g -X -rdynamic to valac command line?

    #1  <unknown>   in 'json_object_get_object_member'
        at /usr/lib/libjson-glib-1.0.so.0
    #2  <unknown>   in 'clone'
        at /usr/lib/libc.so.6
[FATAL 15:56:33.584855] [Json] json_object_get_string_member: assertion 'object != NULL' failed
NULL
An error occured (SIGTRAP)
   The reason is likely an uncaught error.
   Note: no file path and line numbers can be retrieved. Are you sure you added -g -X -rdynamic to valac command line?

    #1  <unknown>   in 'json_object_get_string_member'
        at /usr/lib/libjson-glib-1.0.so.0
    #2  <unknown>   in 'clone'
        at /usr/lib/libc.so.6
[FATAL 15:56:33.689138] [Json] json_object_get_int_member: assertion 'node != NULL' failed
NULL
An error occured (SIGTRAP)
   The reason is likely an uncaught error.
   Note: no file path and line numbers can be retrieved. Are you sure you added -g -X -rdynamic to valac command line?

    #1  <unknown>   in 'json_object_get_int_member'
        at /usr/lib/libjson-glib-1.0.so.0
    #2  <unknown>   in 'clone'
        at /usr/lib/libc.so.6
[FATAL 15:56:33.788396] [Json] json_object_get_object_member: assertion 'node != NULL' failed
NULL
An error occured (SIGTRAP)
   The reason is likely an uncaught error.
   Note: no file path and line numbers can be retrieved. Are you sure you added -g -X -rdynamic to valac command line?

    #1  <unknown>   in 'json_object_get_object_member'
        at /usr/lib/libjson-glib-1.0.so.0
    #2  <unknown>   in 'clone'
        at /usr/lib/libc.so.6
[FATAL 15:56:33.885396] [Json] json_object_get_string_member: assertion 'object != NULL' failed
NULL
An error occured (SIGTRAP)
   The reason is likely an uncaught error.
   Note: no file path and line numbers can be retrieved. Are you sure you added -g -X -rdynamic to valac command line?

    #1  <unknown>   in 'json_object_get_string_member'
        at /usr/lib/libjson-glib-1.0.so.0
    #2  <unknown>   in 'clone'
        at /usr/lib/libc.so.6
[FATAL 15:56:33.986564] [Json] json_object_get_int_member: assertion 'node != NULL' failed
NULL
An error occured (SIGTRAP)
   The reason is likely an uncaught error.
   Note: no file path and line numbers can be retrieved. Are you sure you added -g -X -rdynamic to valac command line?

    #1  <unknown>   in 'json_object_get_int_member'
        at /usr/lib/libjson-glib-1.0.so.0
    #2  <unknown>   in 'clone'
        at /usr/lib/libc.so.6
[FATAL 15:56:34.086801] [Json] json_object_get_object_member: assertion 'node != NULL' failed
NULL
An error occured (SIGTRAP)
   The reason is likely an uncaught error.
   Note: no file path and line numbers can be retrieved. Are you sure you added -g -X -rdynamic to valac command line?

    #1  <unknown>   in 'json_object_get_object_member'
        at /usr/lib/libjson-glib-1.0.so.0
    #2  <unknown>   in 'clone'
        at /usr/lib/libc.so.6
...
```

Using gdb:

```
(gdb) r
Starting program: /usr/bin/gamehub --debug
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fffe55e5700 (LWP 25256)]
Registering Stacktrace 0.9.4
[INFO 15:55:54.071025] Application.vala:154: GameHub version: 0.8.0
[INFO 15:55:54.071049] Application.vala:156: Kernel version: 4.18.4-arch1-1-ARCH
[New Thread 0x7fffe43f7700 (LWP 25257)]
[New Thread 0x7fffe3bd6700 (LWP 25258)]
[New Thread 0x7fffe31c8700 (LWP 25259)]
[INFO 15:55:54.167634] app.vala:32: Distro: NAME="Arch Linux"
[DEBUG 15:55:54.168927] Settings.vala:163: Loading settings from schema 'com.github.tkashkin.gamehub.ui'
[DEBUG 15:55:54.202509] Settings.vala:163: Loading settings from schema 'com.github.tkashkin.gamehub.auth.steam'
...
[New Thread 0x7fffd2ed4700 (LWP 25268)]
[New Thread 0x7fffd26d3700 (LWP 25269)]
[DEBUG 15:55:54.832358] Settings.vala:163: Loading settings from schema 'com.github.tkashkin.gamehub.paths'
[DEBUG 15:55:54.832418] Settings.vala:163: Loading settings from schema 'com.github.tkashkin.gamehub.paths'
[New Thread 0x7fffd1ed2700 (LWP 25270)]
[Thread 0x7fffd1ed2700 (LWP 25270) exited]
[New Thread 0x7fffd1ed2700 (LWP 25271)]
[New Thread 0x7fffd12ff700 (LWP 25272)]
[New Thread 0x7fffcbfff700 (LWP 25277)]
[FATAL 15:55:54.889867] [Json] json_object_get_object_member: assertion 'node != NULL' failed

Thread 8 "HumbleLoading" received signal SIGTRAP, Trace/breakpoint trap.
[Switching to Thread 0x7fffd26d3700 (LWP 25269)]
0x00007ffff797fa96 in ?? () from /usr/lib/libglib-2.0.so.0
(gdb) bt
#0  0x00007ffff797fa96 in  () at /usr/lib/libglib-2.0.so.0
#1  0x00007ffff7980def in g_logv () at /usr/lib/libglib-2.0.so.0
#2  0x00007ffff7980fe0 in g_log () at /usr/lib/libglib-2.0.so.0
#3  0x00007ffff385edbb in json_object_get_object_member () at /usr/lib/libjson-glib-1.0.so.0
#4  0x0000555555578fb9 in  ()
#5  0x0000555555579844 in  ()
#6  0x00005555555768ac in  ()
#7  0x00007ffff79a21ea in  () at /usr/lib/libglib-2.0.so.0
#8  0x00007ffff351ba9d in start_thread () at /usr/lib/libpthread.so.0
#9  0x00007ffff344ba43 in clone () at /usr/lib/libc.so.6
```

**Comment by neuromancer** [comment]
Show non-native games was disable. Strangely enough, it works fine if you enable it. All games are there.

**Comment by tkashkin** [comment]
I'm really confused now. Filtering seems to look right, I haven't changed anything and now it works. :thinking:

**Comment by tkashkin** [comment]
@neuromancer I've tested with your `games.db` and noticed you have way more games on GOG and Humble. :slightly_smiling_face:

I've fixed a couple of Humble-specific bugs which did not occur for me because I have only a few games there.

This issue should be fixed.

Also app should start and be usable much faster for you because HumbleLoading thread was consuming all time and Thread.sleep was placed in callback instead of in actual cycle.
